server/src/main/kotlin/suwayomi/tachidesk/graphql/queries/filter/Filter.kt

```diff
@@ -451,15 +520,8 @@ fun <T : Comparable<T>> andFilterWithCompareEntity(
     column: Column<EntityID<T>>,
     filter: ComparableScalarFilter<T>?,
 ): Op<Boolean>? {
-    filter ?: return null
-    val opAnd = OpAnd(andFilterEntity(column, filter))
-
-    opAnd.andWhere(filter.lessThan) { column less it }
-    opAnd.andWhere(filter.lessThanOrEqualTo) { column lessEq it }
-    opAnd.andWhere(filter.greaterThan) { column greater it }
-    opAnd.andWhere(filter.greaterThanOrEqualTo) { column greaterEq it }
-
-    return opAnd.op
+    @Suppress("UNCHECKED_CAST")
+    return andFilterWithCompare(column as Column<T>, filter)
```

Comment:
Huh, this cast works? Wouldn't it fail because T is wrapped by an entity

Reply:
I assume it works because it has to be a column with a comparable value, which EntityID is

Reply:
This is a simple query to test it, let me know if it works
```gql
query GetManga {
    mangas(filter: { id: { in: [1, 2] } }) {
        id
    }
}
```

Reply:
it's working (I wouldn't have created the pr otherwise 😄)

from googling: it works because of generic type erasure, so at runtime it's just Column<Comparable> to Column<Comparable>